NitroxClient/MonoBehaviours/Discord/DiscordClient.cs

```diff
@@ -64,9 +65,33 @@ private void OnDisable()
         discord?.Dispose();
     }
 
+    private void OnDestroy()
+    {
+        if (main == this)
+        {
+            main = null;
+            activity = default;
+        }
+    }
+
     private void Update()
     {
-        discord?.RunCallbacks();
+        try
+        {
+            discord?.RunCallbacks();
+        }
+        catch (Exception ex)
+        {
+            DiposeAndScheduleHookRestart();
+            Log.ErrorOnce($"An error occured while running callbacks for Discord, will retry every {RETRY_INTERVAL} seconds: {ex.Message}");
+        }
```

Comment:
Is this triggered by the usual payload error we sometimes have?

Reply:
It happens when Discord is closed while Nitrox has its Discord hook running (and for other reason) but it's unrelated to the payload error which is harmless to a certain extent.
Added a comment for it